opapi/src/section-types-generator/section-types-generator.ts

```diff
@@ -0,0 +1,87 @@
+import { initDirectory, saveFile } from 'src/file'
+import {
+  parseFunctionDefinition,
+  parseParameterTypes,
+  parseRequestParameterTypes,
+  parseSectionTypes,
+  parseReturnTypes,
+} from './section-types-generator.parsers'
+import { Block, DefaultState, OperationParser, SectionParser, BlockComposer } from './section-types-generator.types'
+import { camel } from 'radash'
+import { getBlankBlock } from './section-types-generator.helpers'
+
+const operationGenerators: OperationParser[] = [
+  parseFunctionDefinition,
+  parseParameterTypes,
+  parseRequestParameterTypes,
+  parseReturnTypes,
+]
+const sectionGenerators: SectionParser[] = [parseSectionTypes]
+
+/**
+ * Generates files containing typescript types for each item in the state object - Sections, Operations, Responses, etc.
+ */
+export async function generateTypesBySection(state: DefaultState, targetDirectory: string) {
+  initDirectory(targetDirectory)
+  state.sections.forEach(async (section) => {
+    const [sectionBlocks, operationBlocks] = await Promise.all([
+      executeSectionParsers(sectionGenerators, section, state),
+      executeOperationParsers(operationGenerators, section, state),
+    ])
+    composeFilesFromBlocks([...sectionBlocks, ...operationBlocks], targetDirectory)
+  })
+}
+
+const composeFilesFromBlocks: BlockComposer = (blocks: Block[], targetDirectory: string) => {
+  blocks.forEach((block) => {
+    let content = ''
+    content = getImportsForDependencies(block, blocks, content)
+    content += block.content
+    if (Boolean(content) && Boolean(block.title)) {
```

Comment:
**nitpick:** not sure I like this `Boolean(content)` notation, why not just:

```ts
if (content && block.title) { ... }
```

or 

```ts
if (!!content && !!block.title) { ... }
```